src/mina-signer/sign-legacy.ts

```diff
@@ -0,0 +1,248 @@
+import { UInt32, UInt64 } from '../provable/field-bigint.js';
+import { PrivateKey, PublicKey } from '../provable/curve-bigint.js';
+import { HashInputLegacy } from '../provable/poseidon-bigint.js';
+import { Memo } from './memo.js';
+import { NetworkId, Signature, signLegacy, verifyLegacy } from './signature.js';
+import { Json } from '../provable/gen/transaction-bigint.js';
+import { bytesToBits, stringToBytes } from '../provable/binable.js';
+
+export {
+  signPayment,
+  signStakeDelegation,
+  signString,
+  verifyPayment,
+  verifyStakeDelegation,
+  verifyStringSignature,
+  PaymentJson,
+  DelegationJson,
+};
+
+function signPayment(
+  payment: PaymentJson,
+  privateKeyBase58: string,
+  networkId: NetworkId
+) {
+  let command = paymentFromJson(payment);
+  return signUserCommand(command, privateKeyBase58, networkId);
+}
+function signStakeDelegation(
+  delegation: DelegationJson,
+  privateKeyBase58: string,
+  networkId: NetworkId
+) {
+  let command = delegationFromJson(delegation);
+  return signUserCommand(command, privateKeyBase58, networkId);
+}
+
+function signUserCommand(
+  command: UserCommand,
+  privateKeyBase58: string,
+  networkId: NetworkId
+) {
+  let input = toInputLegacy(command);
+  let privateKey = PrivateKey.fromBase58(privateKeyBase58);
+  let signature = signLegacy(input, privateKey, networkId);
+  return Signature.toBase58(signature);
```

Comment:
Does `Signaure.toBase58` have a chance to panic/crash? Do we handle that further down somewhere?

Reply:
I think that toBase58, as opposed to fromBase58, can't crash because the output from `signLegacy` will always be a valid signature. But I'll check!